worker/src/main/java/gov/cms/ab2d/worker/service/BeneficiaryServiceImpl.java

```diff
@@ -0,0 +1,85 @@
+package gov.cms.ab2d.worker.service;
+
+import gov.cms.ab2d.common.model.Beneficiary;
+import gov.cms.ab2d.common.model.Contract;
+import gov.cms.ab2d.common.model.Coverage;
+import gov.cms.ab2d.common.repository.BeneficiaryRepository;
+import gov.cms.ab2d.common.repository.ContractRepository;
+import gov.cms.ab2d.common.repository.CoverageRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.LinkedHashSet;
+import java.util.Optional;
+import java.util.Set;
+
+@Slf4j
+@Service
+@RequiredArgsConstructor
```

Comment:
Do you need @Transactional at the class level here? One of the methods that hits the DB has it, but I'm wondering if the query that reads needs it too?

Reply:
I feel only the insert method needs to be in a transaction.
I do not see any value to adding transactions to the read method.

Reply:
There's some reasons discussed here for using transactions with read e.g. optimizations, but may not make a huge difference here https://stackoverflow.com/a/13540484

Reply:
Thanks for the link. Will study it.